doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -199,53 +222,123 @@ The DHCP_SERVER_IPV4_PORT table includes the following:
 
 - For each DPU available in the system the IP address that should be assigned to the DPU.
 
-### Midplane bridge configuration flow ###
+### Midplane network configuration flow ###
 
-![image](./smart-switch-ip-assignment-midplane-bridge-configuration-flow.png)
+![image](./smart-switch-ip-assignment-midplane-network-configuration-flow.svg)
 
-1. Interface-config service when started consumes configuration from the config DB.
-2. Interface-config service renders interfaces configuration file that includes configuration for midplane bridge.
-3. Networking service consumes the interface configuration and configures the networking subsystem
+#### NPU ####
 
-After the transformations performed by the interface-config service the interfaces configuration will have the following entries:
+1. systemd-sonic-generator renders configuration file of systemd-networkd and midplane-network.service according to the platform.json.
 
+``` text
+# bridge-midplane.netdev
+[NetDev]
+Name=bridge-midplane
+Kind=bridge
 ```
-auto bridge_midplane
-iface bridge_midplane inet static
-    bridge_ports Dpu0 Dpu1
-    address 169.254.200.254
-    netmask 255.255.255.0
-    network 169.254.200.0
-    broadcast 169.254.200.255
+
+``` text
+# bridge-midplane.network
+[Match]
+Name=bridge-midplane
+
+[Network]
+Address=169.254.200.254/24
 ```
 
-### IP assignment flow ###
+``` text
+# midplane-network.network
+[Match]
+Name=dpu0 dpu1
```

Comment:
@Pterosaur How is this file generated for `N` DPUs in a platform?

Reply:
This file will be generated/edited by [sonic-systemd-generator.](https://github.com/sonic-net/sonic-buildimage/tree/master/src/systemd-sonic-generator). And this process will get the DPU information from platform.json.